drainer/checkpoint/mysql.go

```diff
@@ -68,6 +74,16 @@ func newMysql(cfg *Config) (CheckPoint, error) {
 		return nil, errors.Annotatef(err, "exec failed, sql: %s", sql)
 	}
 
+	if sp.clusterID == 0 {
+		id, err := getClusterID(db, sp.schema, sp.table)
```

Comment:
This logic is very weird. 
In order to handle the situation where the upstream cluster is completely suspended, drainer needs a correct way to fetch the  cluster id. But current implementation is not satisfying, like drainer can't allow store multiple entries in checkpoint table

Reply:
I file an issue about it https://github.com/pingcap/tidb-binlog/issues/889